### PR TITLE
Make repo compliant

### PR DESCRIPTION
**Pull request by RaphaelVogel**
**How to categorize this PR?**
/area compliance
/kind cleanup

**What this PR does / why we need it**:
* Created `CONTRIBUTING.md` file
* Created/Updated `LICENSE` file
  * Deleted `LICENSE.md` file
* Created/Updated `.reuse/dep5` file
* Created/Updated `LICENSES` folder
* Added badge for reuse compliance in `README.md` file and registered repo with REUSE API
* Modified `add_license_headers.sh` script
   * added `hack/license_boilerplate.txt` file 
* License/Copyright header replacement for 'non-go' files
  * .ci/common.sh
  * .ci/build
  * hack/add_license_headers.sh
  * hack/build.sh
  * hack/check.sh
  * hack/test.sh
  * hack/tools.mk
  * hack/local-dev/etcd-up.sh
  * hack/local-dev/etcd-down.sh
  * hack/local-dev/kind.sh
  * hack/local-dev/generate_pki.sh
  * hack/local-dev/cleanup.sh
  * hack/local-dev/generate_k8s_resources.sh
  * ops/print-etcd-cheatsheet.sh
* License/Copyright header replacement for 'go' files
  * main.go
  * cmd/command.go
  * cmd/etcd_test.go
  * cmd/etcd.go
  * cmd/help.go
  * internal/types/config.go
  * internal/types/constants.go
  * internal/types/config_test.go
  * internal/bootstrap/loggerutil.go
  * internal/bootstrap/bootstrap.go
  * internal/bootstrap/bootstrap_test.go
  * internal/app/readycheck.go
  * internal/app/fakes.go
  * internal/app/app.go
  * internal/app/readycheck_test.go
  * internal/util/retry_test.go
  * internal/util/http_test.go
  * internal/util/http.go
  * internal/util/retry.go
  * internal/util/tls.go
  * internal/util/tls_test.go
  * internal/testutil/tls.go
  * internal/signal/signal.go
  * internal/signal/signal_test.go
  * internal/brclient/brclient_test.go
  * internal/brclient/brclient.go

**Additional information**
To keep this repository compliant, CODEOWNERS must maintain their third party dependencies in the vendor folder using `.reuse/dep5` file. To prevent constant maintenance effort, it is **highly recommended** to remove the `vendor` folder. In this case no third party dependencies of the vendor folder must be maintained anymore!

### CODE REVIEW

File: hack/license_boilerplate.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
```
Comment (shreyas-s-rao) [inline hack/license_boilerplate.txt:1]: Missing `year`.

While @unmarshall and I were working on [etcd-druid#728](https://github.com/gardener/etcd-druid/issues/728), and were rebasing on latest etcd-druid master, we saw that the year in the boilerplate.txt file was hardcoded by [this PR](https://github.com/gardener/etcd-druid/pull/770) from you. We made a couple of changes to make it work for any given year, in [this commit](https://github.com/unmarshall/etcd-druid/commit/3c550a3a3ee0afa775b087bab39426be2c412044). Please incorporate something similar for this PR as well.

Essentially, you need to first remove the `/`s in the boilerplate file, and use sed to replace the `{YEAR}` with the actual year, like we've done [here](https://github.com/unmarshall/etcd-druid/commit/3c550a3a3ee0afa775b087bab39426be2c412044#diff-bc5efa1d5f2c010c2f712560dead0417d82695f7158075a80ba276abf6003d4c). Then, the Makefile target needs to be enhanced to pass a `YEAR` parameter to the script, like [this](https://github.com/gardener/etcd-druid/blob/9c5f8254e3aeb24c1e3e88d17d8d1de336ce981b/Makefile#L145-L147), and finally run the make target as `YEAR=2024 make add-license-headers`.

```suggestion
// SPDX-FileCopyrightText: {YEAR} SAP SE or an SAP affiliate company and Gardener contributors
```

**Review by shreyas-s-rao — CHANGES_REQUESTED** [review]

File: hack/license_boilerplate.txt (hunk 1 above)
Comment (RaphaelVogel) [reply hack/license_boilerplate.txt:1]: regarding https://github.com/gardener/etcd-druid/pull/770
My intend was to not include the year at all in hack/boilerplate.go.txt ! That's actually a mistake on my side

**Comment by RaphaelVogel** [comment]
@shreyas-s-rao I have implemented your change requests

**Review by shreyas-s-rao — APPROVED** [review]
@RaphaelVogel thanks for making the changes.

/lgtm

**Pull request merged**